python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tracing.ipynb

```diff
@@ -0,0 +1,436 @@
+    "        model_client=model_client,\n",
+    "        system_message=\"\"\"\n",
+    "        You are a planning agent.\n",
+    "        Your job is to break down complex tasks into smaller, manageable subtasks.\n",
+    "        Your team members are:\n",
+    "            WebSearchAgent: Searches for information\n",
+    "            DataAnalystAgent: Performs calculations\n",
+    "\n",
+    "        You only plan and delegate tasks - you do not execute them yourself.\n",
+    "\n",
+    "        When assigning tasks, use this format:\n",
+    "        1. <agent> : <task>\n",
+    "\n",
+    "        After all tasks are complete, summarize the findings and end with \"TERMINATE\".\n",
+    "        \"\"\",\n",
+    "    )\n",
+    "\n",
+    "    web_search_agent = AssistantAgent(\n",
+    "        \"WebSearchAgent\",\n",
+    "        description=\"An agent for searching information on the web.\",\n",
+    "        tools=[search_web_tool],\n",
+    "        model_client=model_client,\n",
+    "        system_message=\"\"\"\n",
+    "        You are a web search agent.\n",
+    "        Your only tool is search_tool - use it to find information.\n",
+    "        You make only one search call at a time.\n",
+    "        Once you have the results, you never do calculations based on them.\n",
+    "        \"\"\",\n",
+    "    )\n",
+    "\n",
+    "    data_analyst_agent = AssistantAgent(\n",
+    "        \"DataAnalystAgent\",\n",
+    "        description=\"An agent for performing calculations.\",\n",
+    "        model_client=model_client,\n",
+    "        tools=[percentage_change_tool],\n",
+    "        system_message=\"\"\"\n",
+    "        You are a data analyst.\n",
+    "        Given the tasks you have been assigned, you should analyze the data and provide results using the tools provided.\n",
+    "        If you have not seen the data, ask for it.\n",
+    "        \"\"\",\n",
+    "    )\n",
+    "\n",
+    "    text_mention_termination = TextMentionTermination(\"TERMINATE\")\n",
+    "    max_messages_termination = MaxMessageTermination(max_messages=25)\n",
+    "    termination = text_mention_termination | max_messages_termination\n",
+    "\n",
+    "    selector_prompt = \"\"\"Select an agent to perform task.\n",
+    "\n",
+    "    {roles}\n",
+    "\n",
+    "    Current conversation context:\n",
+    "    {history}\n",
+    "\n",
+    "    Read the above conversation, then select an agent from {participants} to perform the next task.\n",
+    "    Make sure the planner agent has assigned tasks before other agents start working.\n",
+    "    Only select one agent.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    task = \"Who was the Miami Heat player with the highest points in the 2006-2007 season, and what was the percentage change in his total rebounds between the 2007-2008 and 2008-2009 seasons?\"\n",
+    "\n",
+    "    runtime = SingleThreadedAgentRuntime()\n",
```

Comment:
We don't need to set the `runtime` now as this will be automatically created by the team if not provided. Also currently it's still experimental to customize runtime.